### DISCUSSION TITLE
Processing and Streaming Audio from the Server to the Client

### DISCUSSION THREAD

**Discussion by ophirgal** [aiortc/aiortc#663] [Q&A]
Hi, this looks like a fantastic project,

I'm new to `aiortc` and I'm trying to develop a simple web app that streams audio files from the server's file system to the client, but I want to also process the audio chunks with a deep neural network before streaming them to the client. This is similar to what is done in the 'server' example, but I'm a beginner to `aiortc` and it's not clear to me how I should change the example for my purpose, can anyone give me advice on how to go about this? 

It seems to me I need to subclass `MediaStreamTrack` and process the audio frames inside the `recv` method, but it seems that `recv` receives one audio frame at a time, and my neural network needs audio chunks, e.g. 2048 frames or 4096 frames, so here are two questions:
1. How do I use this class to send a WAV file that's on the server (as opposed to one that's being sent from the client)?
2. How do I get audio chunks instead of frames? Would something like the following work ? --
```
frames = []
for _ in range(2048):
    frames.append(await self.track.recv())

# ... process the audio frames as a 2048-sized chunk ...

# return frames one by one
for frame in frames:
    yield frame
```

Any help would be greatly appreciated, thanks!!

**Comment by jsiqueir-sinch** [comment]
Any luck @ophirgal ?
